src/ray/util/pipe_logger.cc

```diff
@@ -218,22 +149,18 @@ bool ShouldUsePipeStream(const StreamRedirectionOption &stream_redirect_opt) {
 
 #if defined(__APPLE__) || defined(__linux__)
 RedirectionFileHandle OpenFileForRedirection(const std::string &file_path) {
-  int fd = open(file_path.data(), O_WRONLY | O_CREAT, 0644);
-  RAY_CHECK_NE(fd, -1) << "Fails to open file " << file_path << " with failure reason "
-                       << strerror(errno);
-
-  auto flush_fn = [fd]() {
-    RAY_CHECK_EQ(fsync(fd), 0) << "Fails to flush data to disk because "
-                               << strerror(errno);
-  };
-  auto close_fn = [fd]() {
-    RAY_CHECK_EQ(fsync(fd), 0) << "Fails to flush data to disk because "
-                               << strerror(errno);
-    RAY_CHECK_EQ(close(fd), 0) << "Fails to close redirection file because "
-                               << strerror(errno);
+  boost::iostreams::file_descriptor_sink sink{file_path, std::ios_base::out};
+  auto handle = sink.handle();
+  auto ostream =
+      std::make_shared<boost::iostreams::stream<boost::iostreams::file_descriptor_sink>>(
+          std::move(sink));
+  auto flush_fn = [ostream]() { ostream->flush(); };
+  auto close_fn = [ostream]() {
+    ostream->flush();
+    ostream->close();
   };
-
-  return RedirectionFileHandle{fd, std::move(flush_fn), std::move(close_fn)};
+  return RedirectionFileHandle{
+      handle, std::move(ostream), std::move(flush_fn), std::move(close_fn)};
```

Comment:
I think this is the case where we want to work on the fd level. We are not using boost to do cross-platform in this case. Not sure if `ostream->flush()` is the same as `fsync(fd)`.

Reply:
Actually why not changing windows code to this as well?

Reply:
As of now, the ostream that we have is stdout and stderr, hard to imagine it's doing anything else beside flushing content into console.

Reply:
updated the code to reduce unnecessary windows code